cloudformation/external-slurmdbd/resources/user_data.sh

```diff
@@ -7,13 +7,21 @@ function vendor_cookbook
   tar -xzf /etc/chef/aws-parallelcluster-cookbook.tgz
   HOME_BAK="${!HOME}"
   export HOME="/tmp"
-  for d in `ls /tmp/cookbooks`; do
-    cd /tmp/cookbooks/$d
+  for d in /tmp/cookbooks/*; do
+    cd "$d" || continue
     LANG=en_US.UTF-8 /opt/cinc/embedded/bin/berks vendor /etc/chef/cookbooks --delete
   done;
   export HOME="${!HOME_BAK}"
 }
 
+TOKEN=`curl -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600"`
```

Comment:
Nitpick: There is a small time gap between bootsrapping the instance and when we wont have this 2nd IP address assigned to the instance. The user_data here is  not MIME part file so this is one of the last hook which will run before the instance is up and running for us


This is non blocking, however if we add the write_files section and make this a MIME file like we do for HN, CF,etc   
then adding/assigning the IP as part of the bootstrap should be possible.

Reply:
How much is the time gap?

Reply:
Its usually a few seconds, but not sure.

But do we need to check if we check if the IP is available( like a validation) or is that something we dont need to be responsible for?

Reply:
We don' check ip availability. "--allow-reassignment" just seize the ip address for the last call